.travis.yml

```diff
@@ -43,11 +43,12 @@ before_script:
   - for i in {1..4}; do echo "$(sleep 500)" ; done &
   - set -o errexit; source ./ci/test/05_before_script.sh &> "/dev/null"
 script:
+  - if [[ -n "$USE_VALGRIND" || "$RUN_FUNCTIONAL_TESTS" = "true" || "$RUN_FUZZ_TESTS" = "true" ]]; then export LONG_RUNNING_TEST=1; fi
   - export CONTINUE=1
   - if [ $SECONDS -gt 1200 ]; then export CONTINUE=0; fi  # Likely the depends build took very long
   - if [ $TRAVIS_REPO_SLUG = "bitcoin/bitcoin" ]; then export CONTINUE=1; fi  # continue on repos with extended build time (90 minutes)
   - if [ $CONTINUE = "1" ]; then set -o errexit; source ./ci/test/06_script_a.sh; else set +o errexit; echo "$CACHE_ERR_MSG"; false; fi
-  - if [ $SECONDS -gt 2000 ]; then export CONTINUE=0; fi  # Likely the build took very long; The tests take about 1000s, so we should abort if we have less than 50*60-1000=2000s left
+  - if [ $SECONDS -gt 2000 && $LONG_RUNNING_TEST = "1" ]; then export CONTINUE=0; fi  # Likely the build took very long; The tests take about 1000s, so we should abort if we have less than 50*60-1000=2000s left
```

Comment:
Should this be an `&&` or `||` and why? Also, why use the runtime heuristic in combination with the heuristic in line 46. Wouldn't it be easier to simply export `LONG_RUNNING_TEST` for the configs that need it? I.e. right in the line where the other env vars are defined:

```
      env: >-
        FILE_ENV="..."
        LONG_...

Reply:
^ this will not work as the `LONG_RUNNING_TEST` actually should depends on the `TRAVIS_REPO_SLUG`.

Another suggestion is to `export LONG_RUNNING_TEST="true"` by default in `00_setup_env.sh` and set it in a job setup script `00_setup_env_*.sh`:
```
if [ "$TRAVIS_REPO_SLUG" != "bitcoin/bitcoin" ]; then
  export RUN_FUNCTIONAL_TESTS="false"
  export LONG_RUNNING_TEST="false"
fi
```

Reply:
If you don't set `LONG_RUNNING_TEST` to anything for the macos config, everything will work fine in all repos, no?

* `bitcoin/bitcoin` will run the functional tests, and forked repos won't. Independent of `LONG_RUNNING_TEST`

Reply:
> If you don't set `LONG_RUNNING_TEST` to anything for the macos config, everything will work fine in all repos, no?
> 
>  - `bitcoin/bitcoin` will run the functional tests, and forked repos won't. Independent of `LONG_RUNNING_TEST`

With cleared (or failed) cache forked repos could experience timeouts.

The mean of `LONG_RUNNING_TEST` is:
- it is set -- do not continue tests if not enough time remains. This should be done to avoid timeouts in forked repos
- it is unset -- do not take into account the remaining time, just continue. This should be done in cases of manual disabled tests in forked repos.

Actually, the `LONG_RUNNING_TEST` only affects forked repos due to the following code:
https://github.com/bitcoin/bitcoin/blob/4db44acf2d5d3b40943c94b3960f42838255b7ad/.travis.yml#L51-L52

Reply:
Ok, so this seems to affect more than just macos. Hmm. I am not sure if it is expected to fork the repo, run the tests on a commit and then see a green checkmark when the functional tests haven't even been run. Then the commit is proposed to bitcoin/bitcoin and the tests fail.

Excluding macOS is fine because I am not aware this ever detected a bug that was not detected by any of the other builds. Though, disabling all tests for forked repos seems a bit too much, no?

If travis can't give us what we need, we should switch providers instead of disabling tests conditionally.